appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/UberJarCreator.java

```diff
@@ -241,6 +251,19 @@ public void buildUberJar() {
                     }
                 }
             }
+            
+            if (copyDirectory != null) {
+                String basePath = copyDirectory.getPath().replaceAll(copyDirectory + "$", "");
+                List<File> filesToCopy = fillFiles(copyDirectory);
+                for (File file : filesToCopy) {
+                    
+                        JarEntry deploymentEntry = new JarEntry(copyDirectory + file.getPath().replace(basePath, ""));
+                        jos.putNextEntry(deploymentEntry);
+                        Files.copy(file, jos);
```

Comment:
hmm true. I assume the person who asked the question had manually added the entry via zip or something.